### ISSUE TITLE
Issue with paring - "your permission is required to connect your device to TV" required each time

### ISSUE THREAD

**Issue by TristanBrotherton** [mmende/homebridge-samsungtv-control2#71]
Thank you for all your hard work!

I'm having an issue getting this to pair with my new samsung frame tv (2021). I have tried manually pairing with the command suggest in my homebridge logs:

`sudo npx homebridge-samsungtv-control pair2 192.168.167.4 68:72:c3:c5:a7:0a ornpx homebridge-samsungtv-control pair1 192.168.167.4 68:72:c3:c5:a7:0a`

This appears to work. However any time I try and interact with the TV via homebridge the TV pops a dialogue up saying "Your permission is required to connect your device to the TV [allow] [deny]" - I have turned off the security setting in the TV to ask each time, but still it continues to ask, which renders the plugin unfunctional. 

Example output from my homebridge startup

```
[6/29/2021, 3:28:41 PM] [SamsungTVControl] Found device The Frame (QN75LS03AAFXZA), usn: uuid:c9731f69-25be-43af-8e82-3ccfe89fc33c
Preparing Advertiser for 'The Frame A4D3' using bonjour-hap backend!
Starting to advertise 'The Frame A4D3' using bonjour-hap backend!
[6/29/2021, 3:28:41 PM] The Frame A4D3 is running on port 45775.
[6/29/2021, 3:28:41 PM] Please add [The Frame A4D3] manually in Home app. Setup Code: 123-45-678
```

Example of my config

```
{
    "platform": "SamsungTVControl",
    "devices": [
        {
            "usn": "uuid:c9731f69-25be-43af-8e82-3ccfe89fc33c",
            "token": "14970221",
            "name": "The Frame"
        }
    ]
}
```

Example output of:
`$ sudo npx homebridge-samsungtv-control pair2 192.168.167.4 68:72:c3:c5:a7:0a ornpx homebridge-samsungtv-control pair1 192.168.167.4 68:72:c3:c5:a7:0a`

```
Ok... sending the pairing request to your tv. Please click allow when asked
Looks good so far. Sending the MUTE key to your tv to see if it works...
Did the tv switch it's mute state? If yes then add this to your config as token: 14970221

If it didn't work, here are some other possible solutions:
        1. Try another port npx homebridge-samsungtv-control pair2 192.168.167.4 68:72:c3:c5:a7:0a --port 8001
        2. Try another port npx homebridge-samsungtv-control pair2 192.168.167.4 68:72:c3:c5:a7:0a --port 55000
        3. Try the other pairing method npx homebridge-samsungtv-control pair1 192.168.167.4 68:72:c3:c5:a7:0a
        4. Try the legacy protocol npx homebridge-samsungtv-control legacy 192.168.167.4 68:72:c3:c5:a7:0a
```

All other pairing methods suggested fail.

**Comment by psdc-create** [comment]
I have exactly the same problem. 
I also get this message in the log file:

This plugin slows down Homebridge. The write handler for the characteristic 'Active Identifier' on the accessory 'Bedroom TV CBCB' didn't respond at all!. Please check that you properly call the callback! See https://git.io/JtMGR for more info.

**Comment by CharlesMonneron** [comment]
I have the same issue too with a frame 2020.

**Comment by stueybee** [comment]
Same problem as others. Tried all pairing options. Removed reinstalled bridge and plug-ins.

**Comment by TristanBrotherton** [comment]
Has anyone figured / managed a work around to this issue?

**Comment by Althir81** [comment]
yeah would be great.... paring seems to work with my tv, but then nothing... and tv is always askin for permission after a command send from homekit.

**Comment by stueybee** [comment]
> Has anyone figured / managed a work around to this issue?

FWIW I have not found a workaround and have given up at this point with this plug-in/bridge having tried all options and retrying after the recent HOOBS upgrade (4.0.92 ). I will wait and see if a bridge or plug-in upgrade fixes the problem.

**Comment by AndrewReaganM** [comment]
I have the same (or a very similar) issue on my 2017 or 2017 model. Pairing worked (muted, was given a pin) but every time I try and do anything in HK the TV asks for permission. I've tried all the different settings for permissions on the TV.

Any ideas on what might be happening/how we can fix it?

**Comment by stueybee** [comment]
i am honing in on the problem at least for me. I turned on debugging and saw errors from the bridge saying it could not connect to my EERO router. That prompted me to look and indeed all my other bridges are listed in EERO as a device. However this samsungTV control bridge never connects to the EERO. I tried bypassing the EERO but had the same issue with the Xfinitity router which would not allow the SamsungTV to connect. I have tried putting in MAC addresses etc to no avail. All my other Hoobs bridges had no problems connecting to my EERO router and all work fine. I am not sure why this is considered a certified plugin when it has a basic connectivity problem like this.

**Comment by canerdianeh** [comment]
Same problem here. FYI, The PIN provided is not the token - that's the HomeKit device PIN. The mute test works, But it still requests permission every time it tries to do something, and a token is never generated. TV is a 2015 6 Series.

**Comment by dvtoever** [comment]
I have a Samsung Frame as well, exactly the same issues as described above; mute was working but kept asking for permissions. Switched to https://github.com/tavicu/homebridge-samsung-tizen plugin and managed to get on/off with art-mode working as well as the apple-remote for homekit-tv so i can control the volume, up/down buttons etc with the apple remote widget from my iOS widget panel
